### DISCUSSION TITLE
Unzuverlässige Zahlenerkennung

### DISCUSSION THREAD

**Discussion by gimmel** [jomjol/AI-on-the-edge-device#3155] [General]
### The Problem

Ich lese mit der Version 15.7.0 meinen Wasserzähler ein.
1. Leider wird die 0 oft als 8 erkannt und so werden viel zu hohe Werte erzeugt (dig-cont_0620_s3_q.tflite).Mit dig-class11_1700_s2.tflite ist es genau umgekehrt: Die 8 wird zur 0. Bis gestern hatte die Erkennung aber noch relativ gut funktioniert.
2. Wenn ich zum Testen ein anderes Model wähle, wird nach einem Neustart der Previous Value vergessen. Deshalb wird trotz Maximum Rate Value ein völlig falscher Wert in HomeAssistant geschrieben. (878.3920 statt 78.3829)

### Version

15.7.0

### Logfile

```shell
[0d00h21m01s] 2024-07-25T11:50:42 <ERR> [POSTPROC] main: Raw: 00878.2412, Value: , Status: Rate too high - Read: 878.2412 - Pre: 78.2412 - Rate: 800.0000
```

### Expected Behavior

Null sollte als Null erkannt werden, nicht als Acht.
Acht sollte nicht als Null erkannt werden.

### Screenshots

<img width="1020" alt="Bildschirmfoto 2024-07-25 um 22 35 38" src="https://github.com/user-attachments/assets/793aa23f-2e9c-4546-9ef0-04a9b3522801">
<img width="1029" alt="Bildschirmfoto 2024-07-25 um 23 12 41" src="https://github.com/user-attachments/assets/425e8d8a-4102-4c4e-8512-4f28dfb8994b">

**Comment by SybexX** [comment]
Bild unscharf, muß richtig fokussiert werden
---------------------------------------------
https://jomjol.github.io/AI-on-the-edge-device-docs/Reference-Image/#focus
![Unbenannt2](https://github.com/user-attachments/assets/dd2e4ff6-0e03-410f-a877-35feb9ea45fe)

Bildspiegelungen in der Mitte der Zahlen erschweren das Erkennen einer 0 und 8
-----------------------------------------------------------------------------------
https://github.com/jomjol/AI-on-the-edge-device/discussions/1949
![Unbenannt](https://github.com/user-attachments/assets/6ecd752a-5f2f-4d1f-b68f-a39785eb3d18)

Die digitalen ROIs sind nicht richtig gesetzt
--------------------------------------------
https://jomjol.github.io/AI-on-the-edge-device-docs/ROI-Configuration/
![Unbenannt1](https://github.com/user-attachments/assets/4c829f24-e030-4187-bcff-4b134218f312)

Bitte den Inhalt der Links gründlich durchlesen und befolgen, dann sollten die Probleme beseitigt sein.

**Comment by gimmel** [comment]
Die Screenshots waren vom Overview-Screen, der ist immer etwas unschärfer und er zeigt auch die innere Begrenzung der ROI nicht an. Eigentlich ist der Focus ganz gut eingestellt und die Analog-Erkennung funktioniert auch sehr gut. Aber ich habe die Begrenzung der digitalen ROI nun noch etwas enger eingestellt und siehe da - die Erkennung funktioniert jetzt sehr viel besser.

Vielen Dank für die Hilfe und die Links. Mir ist dabei zum ersten mal klar geworden, dass die Dokumentation nicht bei "1.2.3 Configuration Interface" endet, sondern über das Hamburger-Menü oben links weitere Bereiche aufgerufen werden können.

**Comment by SybexX** [comment]
Du könntest die rolling Version ausprobieren https://github.com/jomjol/AI-on-the-edge-device/actions/runs/10028173125 ,
dort wurden viele Verbesserungen durchgeführt. Mit der neuen zoom-Funktion könntest du, durch vergrößern der ROIs(indem du reinzoomst), die Erkennung auch verbessern.

**Comment by henrythasler** [comment]
Eine weitere Optimierungsmöglichkeit ist, die ESP32CAM-Platine/Halterung bereits so auszurichten, dass das Bild nicht mehr per Software gedreht werden muss und Aliasing-Effekte vermieden werden können.
